daweb/js2/organizace-bundle/vite.md

```diff
@@ -0,0 +1,103 @@
+## Vite
+
+Ve chvili, kdy máme založený projekt, budeme potřebovat nástroj, který vezme náš kód spolu s použitými knihovnami a zabalí je všechny do jednoho velkého JavaScriptového souboru. Ten si pak prohlížeč může stáhnout celý najednou a nemusí tak čekat na spoustu malých souborů jednotlivě po sobě. Nástrojům, které toto pro nás dělají se říká _bundlery_. Existuje jich několik různých. Jeden z nejpopulárnější se jmenuje [_Webpack_](https://webpack.js.org/). My v tomto kurzu budeme používat modernější bundler jménem [_Vite_](https://vitejs.dev/).
+
+Nejdříve musíme do našeho projektu Vite nainstalovat. To provedeme příkazem
+
+```sh
+$ npm install -D vite
+```
+
+Po tomto zásahu bude náš `package.json` vypadat následovně.
+
+```json
+{
+  "name": "muj-projekt",
+  "version": "0.0.1",
+  "dependencies": {
+    "dayjs": "^1.10.4",
+    "validator": "^13.5.2"
+  },
+  "devDependencies": {
+    "vite": "^4.4.5"
+  }
+}
+```
+
+Přibyla nám nová položka `devDependencies`. V té je vypsán seznam balíčků, které představují takzvané _vývojové závislosti_. Tyto balíčky se nestávají součástí naší výsledné stránky, používáme je pouze během vývoje.
+
+### Struktura projektu
+
+Od této chvíle bude každý náš projekt mít více souborů, než jsme zvyklí. Je tedy dobré projekt smysluplně organizovat, abychom se v něm vyznali.
+
+Nejprve v projektu `muj-projekt` vytvoříme složku `src`. Do té budeme vždy vkládat zdrojové kódy naší aplikace, tedy všechny soubory s JavaScriptem a CSS. Soubor `index.html` se vkládá mimo složku `src`. Zatím tedy v této složce vytvoříme hlavní soubory `index.js` a `index.html`. Struktura souborů pak bude vypadat takto.
+
+```
```

Comment:
Kódim zase vrátilo automatickou detekci jazyku, takže to dělá nepořádek. Pokud změna kodim byl záměr, tak tady by mělo být `text`.